pulsar-broker/src/test/java/org/apache/pulsar/client/api/NonDurableSubscriptionTest.java

```diff
@@ -94,6 +94,53 @@ public void testNonDurableSubscription() throws Exception {
 
     }
 
+    @Test
+    public void testSameSubscriptionNameForDurableAndNonDurableSubscription() throws Exception {
+        String topicName = "persistent://my-property/my-ns/same-sub-name-topic";
+        // 1 setup producer and consumer
+        @Cleanup
+        Producer<String> producer = pulsarClient.newProducer(Schema.STRING).topic(topicName)
+                .create();
+
+        @Cleanup
+        Consumer<String> consumer = pulsarClient.newConsumer(Schema.STRING).topic(topicName)
+                .readCompacted(true)
+                .subscriptionMode(SubscriptionMode.Durable)
+                .subscriptionType(SubscriptionType.Exclusive)
+                .subscriptionName("mix-subscription")
+                .subscriptionInitialPosition(SubscriptionInitialPosition.Earliest)
+                .subscribe();
+
+        // 2 send 10 messages
```

Comment:
why do we need to send all these messages ?

isn't it enough to only create a Consumer and a Reader with the same subscription name ?

we can also make it simpler and more explicit to create:
- 1 Consumer with SubscriptionMode.Durable
- 1 Consumer with SubscriptionMode.NonDurable

Reply:
Oh, yes, you're right.
This test is firstly used for showing the behaviour before this pull request, at that point the reader start from the eariliest only can read 5 messages.
After this pull request, it seems no need to keep this logic.